### ISSUE TITLE
I did not get any bounding box after training 2 days.

### ISSUE THREAD

**Issue by HuangBo-Terraloupe** [yhenon/keras-frcnn#104]
After training 2 days, my mode does not detect any things. But the losses are pretty low.
I print the boxes, they are all None []. When I run the test code, it does not generate any bounding boxes.
Dose someone also has the same experience ?

thanks

**Comment by HuangBo-Terraloupe** [comment]
Did someone tried this network on another dataset? 
For VOCdataset I got the right predictions.

![0](https://user-images.githubusercontent.com/26878250/28121878-c1ebd858-671c-11e7-900f-f2997553a65e.png)

But the network detect nothing for my data. It should look at this: (label)
![roof_detection](https://user-images.githubusercontent.com/26878250/28121901-decede8e-671c-11e7-803d-a3a53254879f.png)

**Comment by yhenon** [comment]
what kind of losses are you getting  in training ?

**Comment by HuangBo-Terraloupe** [comment]
Hello,
@yhenon 
Thanks for the quick answer,
I already found my bug, I apply a wrong class mapping for the background.
And I will retrain the network.
I have another 2 questions: 
1, We should not input any images, which does not contain any object, into the network, right? If a image is totally background, we just ignore them.
2, According to your experiences, if the bounding box is very small, will the training also works? Just like the  Solar_Thermal_Panel on the roof.

thanks

**Comment by ftiguidou** [comment]
Hello,
@HuangBo-Terraloupe 
I am having the same issue, no bounding boxes after training on my own classes and dataset. What was your wrong mapping problem ? Did you change it from the one provided in the original code ?

thanks

**Comment by HuangBo-Terraloupe** [comment]
Hello @ftiguidou 

I write my own function to feed the data into network. I make my data the same format as VOC dataset, and the feed it to network. One thing you to do is, you should map the background to the last class. When I map the background to class 0, the classifier will classify the background, which does not needed. 

hope this will help you.

**Comment by ftiguidou** [comment]
Thanks for your answer,

I also wrote my own function to feed the data into the network. Indeed, I'm working with a mongodb for storing my pictures (I have a very large set) and then I made my own parser which provide the same dictionary that simple_parser.py or pascal_voc_parser.py except for the "filepath" key which is replaced by a "data" key corresponding to the (X, Y, 3) numpy array of the picture. I also changed the cv2.imread function in the data_augment.py augment function by just taking the value stored with the "data" key.

As I said, I have no bounding box provided even if the training process seems to go well (the only exception is the "a must be non-empty" mentioned in another issue). It seems my problem is different from yours as I kept the simple_parser code sample making the bg class last :+1: 
`    if found_bg:
        if class_mapping['bg'] != len(class_mapping) - 1:
            key_to_switch = [key for key in class_mapping.keys() if class_mapping[key] == len(class_mapping) - 1][0]
            val_to_switch = class_mapping['bg']
            class_mapping['bg'] = len(class_mapping) - 1
            class_mapping[key_to_switch] = val_to_switch`

I will try to find out what's going wrong with my experiments and will open an issue if I can't find any answer.

**Comment by Abhijit-2592** [comment]
@HuangBo-Terraloupe  It works for me on other datasets if I use VGG-16. I retried training using inception-v3 and faced the same problem.

**Comment by HuangBo-Terraloupe** [comment]
@Abhijit-2592 
The problem for me is the bbox_threshold.
If the classification score is below the threshold, the bounding box will be filtered.

**Comment by Abhijit-2592** [comment]
@HuangBo-Terraloupe  my prob was with the batch normalization bug when wrapped with a time distributed wrapper in keras. I fixed it by updating my kearas from version 2.0.6 to 2.1.1
